src/main/java/org/interledger/cryptoconditions/Fulfillment.java

```diff
@@ -1,13 +1,36 @@
 package org.interledger.cryptoconditions;
 
-public interface Fulfillment {
+/**
+ * An implementation of a crypto-conditions Fulfillment.
+ *
+ * @see "https://datatracker.ietf.org/doc/draft-thomas-crypto-conditions/"
+ */
+public interface Fulfillment<C extends Condition> {
 
-  ConditionType getType();
+  /**
+   * Accessor for the type of this fulfillment.
+   *
+   * @return A {@link CryptoConditionType} for this fulfillment.
+   */
+  CryptoConditionType getType();
 
-  byte[] getEncoded();
```

Comment:
Not sure if I actually like what I'm about to say, but what if we introduced a method like this on both `Condition` and `Fulfillment`, or one of their subclasses:

```
public Key asKey(){
   return CryptoConditionWriter.asKey(this);
}
```

It's not clear to me that a preimage or prefix condition can mirror the `Key` interface (or even overtly extend it) but the RSA and Ed25519 keys probably could, no?

I guess I'm unclear if you're merely wanting something that's _close_ to a key, but not fully there, or if you're looking for Conditions/Fulfillments to actual be able to be translated into a `Key` and/or passed around as a `Key`. Can you clarify that?